core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala

```diff
@@ -104,12 +104,8 @@ TimeSeriesShard(dataset, storeConfig, shardNum, bufferMemoryManager, rawStore, m
             rawStore.readRawPartitions(dataset, allDataCols, multiPart, computeBoundingMethod(pagingMethods))
               // NOTE: this executes the partMaker single threaded.  Needed for now due to concurrency constraints.
               // In the future optimize this if needed.
-              .mapAsync { rawPart =>
-                partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool).asyncBoundary
-              }
+              .mapAsync { rawPart => partitionMaker.populateRawChunks(rawPart) }
```

Comment:
One more thing, mapAsync (and the fact of using a Task for populateRawChunks) is not needed, just use map.